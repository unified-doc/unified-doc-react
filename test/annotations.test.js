import React from 'react';
import renderer from 'react-test-renderer';

import { htmlContent } from './fixtures';
import { Doc } from '..';

describe('annotations', () => {
  it('does not annotate with empty annotations', () => {
    const annotations = [];
    const tree = renderer
      .create(
        <Doc
          annotations={annotations}
          content={htmlContent}
          filename="doc.html"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not annotate with annotations out of range', () => {
    const annotations = [{ id: 'a', start: 200, end: 400 }];
    const tree = renderer
      .create(
        <Doc
          annotations={annotations}
          content={htmlContent}
          filename="doc.html"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('annotates text nodes (no-overlap)', () => {
    const annotations = [
      { id: 'a', start: 0, end: 4 },
      { id: 'b', start: 5, end: 12 },
    ];
    const tree = renderer
      .create(
        <Doc
          annotations={annotations}
          content={htmlContent}
          filename="doc.html"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies custom class names and styles', () => {
    const annotations = [
      { id: 'a', classNames: ['class-a'], style: {background: 'red', color: 'yellow' }, start: 0, end: 4 },
      { id: 'b', classNames: ['class-b1', 'class-b2'], start: 5, end: 12 },
    ];
    const tree = renderer
      .create(
        <Doc
          annotations={annotations}
          content={htmlContent}
          filename="doc.html"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('annotates text nodes (overlapping)', () => {
    const annotations = [
      { id: 'a', classNames: ['class-a'], start: 0, end: 7 },
      { id: 'b', classNames: ['class-b'], start: 5, end: 10 },
      { id: 'c', classNames: ['class-c'], start: 6, end: 15 },
    ];
    const tree = renderer
      .create(
        <Doc
          annotations={annotations}
          content={htmlContent}
          filename="doc.html"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('TODO: will be implemented. supports annotation callbacks', () => {});
});
