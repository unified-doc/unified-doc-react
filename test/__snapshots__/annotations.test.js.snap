// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`annotations annotates text nodes (no-overlap) 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        <div>
          <mark>
            some
          </mark>
        </div>
      </strong>
      <div>
        

        <mark>
          content
        </mark>
      </div>
    </blockquote>
  </div>
</div>
`;

exports[`annotations annotates text nodes (overlapping) 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        <div>
          <mark>
            some
          </mark>
        </div>
      </strong>
      <div>
        <mark>
          

        </mark>
        <mark>
          <mark>
            c
          </mark>
        </mark>
        <mark>
          <mark>
            <mark>
              o
            </mark>
          </mark>
        </mark>
        <mark>
          <mark>
            nte
          </mark>
        </mark>
        <mark>
          nt
        </mark>
      </div>
    </blockquote>
  </div>
</div>
`;

exports[`annotations applies custom classe names 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        <div>
          <mark>
            some
          </mark>
        </div>
      </strong>
      <div>
        

        <mark>
          content
        </mark>
      </div>
    </blockquote>
  </div>
</div>
`;

exports[`annotations does not annotate with annotations out of range 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        some
      </strong>
      
content
    </blockquote>
  </div>
</div>
`;

exports[`annotations does not annotate with empty annotations 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        some
      </strong>
      
content
    </blockquote>
  </div>
</div>
`;
